sdk/core/Azure.Core/src/ResourceIdentifier.cs

```diff
@@ -197,7 +197,8 @@ private T GetValue<T>(ref T value)
 
         private static ResourceType ChooseResourceType(ReadOnlySpan<char> resourceTypeName, ResourceIdentifier parent, out SpecialType specialType)
         {
-            if (resourceTypeName.Equals(ResourceGroupKey.AsSpan(), StringComparison.OrdinalIgnoreCase))
+            //resourceGroups type is Microsoft.Resources/resourceGroups only when its parent is Subscription
+            if (resourceTypeName.Equals(ResourceGroupKey.AsSpan(), StringComparison.OrdinalIgnoreCase) && parent.ResourceType == ResourceType.Subscription)
```

Comment:
Would it be faster to do the ResourceType check first?

Reply:
the typename is smaller than resourcetype of parent so it will quickly eliminate most and not have to do the second check.